### PR TITLE
Maid 1055

### PR DESCRIPTION
**Pull request by inetic**
Ben, pls, suggest what's missing for this task (or whether to move to another task).

### CODE REVIEW

File: src/routing_membrane.rs [hunk 1]
```diff
@@ -514,6 +518,26 @@ impl<F> RoutingMembrane<F> where F: Interface {
...
+        for peer in peers {
+            self.send(self.name_to_endpoint(peer).into_iter(), &serialised_msg);
+        }
+    }
+
+    fn name_to_endpoint(&self, name: &NameType) -> Vec<&Endpoint> {
+        self.relay_map.get_endpoints(&name).iter()
+            .flat_map(|&&(_, ref endpoint_set)| endpoint_set)
+            .collect()
+
```
Comment (inetic) [inline src/routing_membrane.rs]: I think I'll need to search the `routing_table` here as well (wasn't sure when I first wrote it).

File: src/routing_membrane.rs [hunk 2]
```diff
@@ -514,6 +519,26 @@ impl<F> RoutingMembrane<F> where F: Interface {
         }
     }
 
+    fn send_by_name<'a, I>(&self, peers: I, serialised_msg: Bytes) where I: Iterator<Item=&'a NameType> {
+        for peer in peers {
+            self.send(self.name_to_endpoint(peer).into_iter(), &serialised_msg);
+        }
+    }
+
+    fn name_to_endpoint(&self, name: &NameType) -> Vec<&Endpoint> {
```
Comment (benjaminbollen) [inline src/routing_membrane.rs]: Do we need the name-to endpoint in general?  I only used explicitly the routing table or the relay_map when looking up in this direction.  I explicitly did not want to port this function.  Sorry, didnt tell you this

Comment (benjaminbollen) [inline src/routing_membrane.rs]: relay_map already has a lookup function; so has routing_table.  So depending on where you call it; you should know from the context that you only want to search through relay map exclusive or routing_table.  From the message handlers you should only ever search the routing table and forward inside the SAFE network.  No handler should send to a relay endpoint - I wrote all those

Comment (inetic) [inline src/routing_membrane.rs]: OK, I'll change that

**Comment by benjaminbollen** [comment]
thanks. merged in maidsafe

**Pull request closed without merging**